### ISSUE TITLE
Dockerfile for building proxy

### ISSUE THREAD

**Issue by 2011aad** [aeraki-mesh/meta-protocol-proxy#124]
Same as https://github.com/aeraki-mesh/meta-protocol-proxy/issues/112.
Is there any instructions?
I'm also stuck here.

**Comment by zhaohuabing** [comment]
@2011aad 

Did you try this? https://github.com/aeraki-mesh/meta-protocol-proxy?tab=readme-ov-file#build-metaprotocol-proxy-using-docker

**Comment by 2011aad** [comment]
Following this guide, I build the envoy binary success. But I don’t know what should put into the image, only the envoy binary? Or with some dependencies and configurations. Really appreciate for the reply^_^

**Comment by zhaohuabing** [comment]
You mean how to build istio-proxy？

**Comment by 2011aad** [comment]
Yes. I see when using istio with aeraki, I need change proxy image to meta-protocol-proxy:1.4.2. I want to know how to build a new version.

**Comment by 2011aad** [comment]
Is the following method the right way?
https://discuss.istio.io/t/building-istio-with-custom-envoy/7239

**Comment by zhaohuabing** [comment]
@2011aad  Sorry this hasn't been written down in docs. You can find the script here: https://github.com/aeraki-mesh/istio/blob/aeraki-1.18.1/build-proxy.sh

**Comment by 2011aad** [comment]
Thanks. I'm building istio-proxy with my own envoy binary. After build istio docker.proxyv2, I use the following dockerfile content to replace envoy binary within the image and works. ^_^

## Dockerfile
FROM localhost:5000/proxyv2:latest
COPY envoy /usr/local/bin/envoy
ENTRYPOINT ["/usr/local/bin/pilot-agent"]

**Comment by zhaohuabing** [comment]
Yes, that would also work, but the image will be larger.

**Comment by zhaohuabing** [comment]
Implemented in #130
